src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -0,0 +1,125 @@
+package de.zalando.aruha.nakadi.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.exceptions.NakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import de.zalando.aruha.nakadi.service.subscription.StreamParameters;
+import de.zalando.aruha.nakadi.service.subscription.SubscriptionOutput;
+import de.zalando.aruha.nakadi.service.subscription.SubscriptionStreamer;
+import de.zalando.aruha.nakadi.service.subscription.SubscriptionStreamerFactory;
+import java.io.IOException;
+import java.io.OutputStream;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+import org.zalando.problem.Problem;
+
+@RestController
+public class SubscriptionStreamController {
+    private static final Logger LOG = LoggerFactory.getLogger(SubscriptionStreamController.class);
+
+    private final SubscriptionStreamerFactory subscriptionStreamerFactory;
+    private final ObjectMapper jsonMapper;
+
+    @Autowired
+    public SubscriptionStreamController(final SubscriptionStreamerFactory subscriptionStreamerFactory, final ObjectMapper objectMapper) {
+        this.subscriptionStreamerFactory = subscriptionStreamerFactory;
+        this.jsonMapper = objectMapper;
+    }
+
+    private class SubscriptionOutputImpl implements SubscriptionOutput {
+        private boolean headersSent;
+        private final HttpServletResponse response;
+        private final OutputStream out;
+
+        SubscriptionOutputImpl(final HttpServletResponse response, final OutputStream out) {
+            this.response = response;
+            this.out = out;
+            this.headersSent = false;
+        }
+
+        @Override
+        public void onInitialized() throws IOException {
+            if (!headersSent) {
+                headersSent = true;
+                response.setStatus(HttpStatus.OK.value());
+                response.setContentType("application/x-json-stream");
+                out.flush();
+            }
+        }
+
+        @Override
+        public void onException(final Exception ex) {
+            LOG.warn("Exception occurred while streaming", ex);
+            if (!headersSent) {
+                headersSent = true;
+                try {
+                    if (ex instanceof NakadiException) {
```

Comment:
I'm not sure that it will be correct to return only 400 for Nakadi Exception and 503 for everything else. 
I think that in reality there are more options.